HLTrigger/Muon/plugins/HLTMuonTrkFilter.cc

```diff
@@ -96,11 +100,20 @@ bool HLTMuonTrkFilter::hltFilter(edm::Event& iEvent,
   std::vector<unsigned int> filteredMuons;
   for (unsigned int i = 0; i < muons->size(); ++i) {
     const reco::Muon& muon(muons->at(i));
+    reco::RecoChargedCandidateRef cand(cands, i);
+    reco::TrackRef tk = cand->track();
```

Comment:
Is it guaranteed that this ref to a track is always valid? (or is a check needed here?)

Also, looks like both could be `const`.

Reply:
Added a check just in case, and made them `const`.